src/Microsoft.Data.SqlClient/tests/FunctionalTests/SslOverTdsStreamTest.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.Tests
+{
+    public static class SslOverTdsStreamTest
+    {
+        [Theory]
+        [InlineData(0),InlineData(3),InlineData(128), InlineData(2048), InlineData(8192)]
+        public static void ReadWrite(int readLimit)
+        {
+            const int EncaspulatedPacketCount = 4;
+            const int PassThroughPacketCount = 5;
+
+            SyncTest(EncaspulatedPacketCount, PassThroughPacketCount, readLimit);
+            SyncCoreTest(EncaspulatedPacketCount, PassThroughPacketCount, readLimit);
```

Comment:
Another thing, can we have these as separate 4 tests instead of one single test? If one fails we do not want the rest fails.

Reply:
Split, which caused the arg format to change so i made a TheoryData to pass them.